src/utils/externalLinkHelper.js

```diff
@@ -0,0 +1,36 @@
+import { useState } from 'react';
+
+const useExternalLinkState = () => {
+  const [externalLink, setExternalLink] = useState();
+
+  const handleAllowedExternalLink = (linkTo, dispatch, triggerCodeRun) => {
+    setExternalLink(linkTo);
+    dispatch(triggerCodeRun());
+  };
+
+  const handleRegularExternalLink = (linkTo, setPreviewFile, setExternalLink, dispatch, triggerCodeRun) => {
```

Comment:
`setExternalLink` currently fails on external links because you're not passing it (and actually don't want to! The others can easily be removed with a little rejig
```suggestion
  const handleRegularExternalLink = (linkTo, setPreviewFile) => {
```